nixos/modules/services/backup/burpServer.nix

```diff
@@ -0,0 +1,83 @@
+{ config, lib, networking, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.burpServer;
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.burpServer = {
+
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable burp.";
+      };
+
+      user = mkOption {
+        default = "burp";
+        description = "The user to run burp as";
+      };
+
+      group = mkOption {
+        default = cfg.user;
+        description = "The group to run burp as (default same as user)";
+      };
+
+      configFile = mkOption {
+        default = "/etc/burp/burp.conf";
+        description = "Path to the burp config file. You need to provide a fully working configuration file and setup all paths yourself.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.burp;
+        example = literalExample "pkgs.burp";
+        description = ''burp package to use.'';
+      };
+
+    };
+
+  };
+
+  ###### implementation
+
+  config = mkIf config.services.burpServer.enable {
+
+    users.extraUsers.burp = {
```

Comment:
As for the package options, I'm not fully convinced it is better than using `system.stateVersion` (at least for the server), but I'm more than happy to defer to your judgement, I don't use the service after all, so these are just general comments.